packages/share/lib/share.dart

```diff
@@ -51,4 +52,128 @@ class Share {
+  }) {
+    assert(file != null);
+    assert(file.existsSync());
+
+    return shareFiles(
+      <File>[file],
+      mimeTypes: <String>[mimeType ?? _mimeTypeForFile(file)],
+      subject: subject,
+      text: text,
+      sharePositionOrigin: sharePositionOrigin,
+    );
+  }
+
+  /// Summons the platform's share sheet to share multiple files.
+  ///
+  /// Wraps the platform's native share dialog. Can share a file.
+  /// It uses the `ACTION_SEND` Intent on Android and `UIActivityViewController`
+  /// on iOS.
+  ///
+  /// The optional `sharePositionOrigin` parameter can be used to specify a global
+  /// origin rect for the share sheet to popover from on iPads. It has no effect
+  /// on non-iPads.
+  ///
+  /// May throw [PlatformException] or [FormatException]
+  /// from [MethodChannel].
+  static Future<void> shareFiles(
+    List<File> files, {
+    List<String> mimeTypes,
+    String subject,
+    String text,
+    Rect sharePositionOrigin,
+  }) {
+    assert(files != null);
+    assert(files.isNotEmpty);
+    assert(files.every((file) => file.existsSync()));
+    final Map<String, dynamic> params = <String, dynamic>{
+      'paths': files.map((file) => file.path).toList(),
+      'mimeTypes':
+          mimeTypes ?? files.map((file) => _mimeTypeForFile(file)).toList(),
+    };
+
+    if (subject != null) params['subject'] = subject;
+    if (text != null) params['text'] = text;
+
+    if (sharePositionOrigin != null) {
+      params['originX'] = sharePositionOrigin.left;
+      params['originY'] = sharePositionOrigin.top;
+      params['originWidth'] = sharePositionOrigin.width;
+      params['originHeight'] = sharePositionOrigin.height;
+    }
+
+    return channel.invokeMethod('shareFiles', params);
+  }
+
+  static String _mimeTypeForFile(File file) {
+    assert(file != null);
+    final String path = file.path;
+
+    final int extensionIndex = path.lastIndexOf("\.");
+    if (extensionIndex == -1 || extensionIndex == 0) {
+      return null;
+    }
+
+    final String extension = path.substring(extensionIndex + 1);
+    switch (extension) {
+      // image
+      case 'jpeg':
+      case 'jpg':
+        return 'image/jpeg';
+      case 'gif':
+        return 'image/gif';
+      case 'png':
+        return 'image/png';
+      case 'svg':
+        return 'image/svg+xml';
+      case 'tif':
+      case 'tiff':
+        return 'image/tiff';
+      // audio
+      case 'aac':
+        return 'audio/aac';
+      case 'oga':
+        return 'audio/ogg';
+      // video
+      case 'avi':
+        return 'video/x-msvideo';
+      case 'mpeg':
+        return 'video/mpeg';
+      case 'ogv':
+        return 'video/ogg';
+      // other
+      case 'csv':
+        return 'text/csv';
+      case 'htm':
+      case 'html':
+        return 'text/html';
+      case 'json':
+        return 'application/json';
+      case 'pdf':
+        return 'application/pdf';
+      case 'txt':
+        return 'text/plain';
+    }
+    return null;
```

Comment:
Returning null when unknown seems to go against the spec, [MDN says](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#Discrete_types):

> Generic binary data (or binary data whose true type is unknown) is `application/octet-stream`.

(But if you replace this by the mime library they probably handle that themselves)

Reply:
Checked what happens with `mime` package and that one returns null. So adjusting my code to return `application/octet-stream` in case `mime` returns null.